Source/SwiftLintBuiltInRules/Rules/Style/StatementPositionRule.swift

```diff
@@ -36,175 +20,217 @@ struct StatementPositionRule: CorrectableRule {
-        let filterRanges = Self.uncuddledMatchFilter(contents: file.stringView, syntaxMap: syntaxMap)
-        let validMatches = matches.compactMap(validator).filter(filterRanges)
-                  .filter { file.ruleEnabled(violatingRanges: [$0.range], for: self).isNotEmpty }
-        if validMatches.isEmpty {
-            return 0
-        }
-        for match in validMatches.reversed() {
-            let range1 = match.range(at: 1)
-            let range2 = match.range(at: 3)
-            let newlineRange = match.range(at: 2)
-            var whitespace = contents.bridge().substring(with: range1)
-            let newLines: String
-            if newlineRange.location != NSNotFound {
-               newLines = contents.bridge().substring(with: newlineRange)
-            } else {
-                newLines = ""
+            guard needsCorrection else { return nil }
+
+            // Skip correction if there are comments between brace and keyword
+            guard !validation.hasCommentsBetween else { return nil }
+
+            numberOfCorrections += 1
+
+            if configuration.statementMode == .default {
+                // Update the right brace trailing trivia
+                var newBody = node.body
+                newBody.rightBrace = newBody.rightBrace.with(\.trailingTrivia, .space)
+                let newNode = node.with(\.body, newBody)
+
+                // Update the else keyword leading trivia
+                return newNode.with(\.elseKeyword, elseKeyword.with(\.leadingTrivia, []))
             }
-            if !whitespace.hasPrefix("\n") && newLines != "\n" {
-                whitespace.insert("\n", at: whitespace.startIndex)
+            // Uncuddled mode
+            let newTrivia = Trivia.newline + .spaces(validation.expectedIndentation)
+
+            // Update the right brace trailing trivia
+            var newBody = node.body
+            newBody.rightBrace = newBody.rightBrace.with(\.trailingTrivia, [])
+            let newNode = node.with(\.body, newBody)
+
+            // Update the else keyword leading trivia
+            return newNode.with(\.elseKeyword, elseKeyword.with(\.leadingTrivia, newTrivia))
+        }
+
+        private func correctCatchStatement(catchClause: CatchClauseSyntax, keyword: TokenSyntax) -> CatchClauseSyntax? {
+            guard let previousToken = keyword.previousToken(viewMode: .sourceAccurate),
+                  previousToken.tokenKind == .rightBrace else { return nil }
+
+            let validation = StatementValidation(keyword: keyword, previousToken: previousToken)
+            let needsCorrection = configuration.statementMode == .default ?
+                !validation.isValidForDefaultMode() :
+                !validation.isValidForUncuddledMode()
+
+            guard needsCorrection else { return nil }
+
+            // Skip correction if there are comments between brace and keyword
+            guard !validation.hasCommentsBetween else { return nil }
```

Comment:
Could go into a method in one or the other way to be used here, in `correctIfStatement` and `validateStatement` in the visitor.